extruct/w3cmicrodata.py

```diff
@@ -62,23 +63,47 @@ def __init__(self, nested=True, strict=False, add_text_content=False, add_html_n
         self.add_text_content = add_text_content
         self.add_html_node = add_html_node
 
-    def get_docid(self, node):
-        return int(self._xp_item_docid(node))
+    def get_docid(self, node, itemids):
+        try:
+            return itemids[node]  # same as self.get_docid(node, {})
+        except KeyError:
+            # Even after itemids are built,
```

Comment:
Alternative: compute always all ids starting from the root in `_build_itemids`. According to your comment, this will be the most common case, and for the case in which only a section of the document is extracted, I have the feeling that the speed would be equivalent or faster.  But probably code will simplify a little bit as `_xp_item_docid` wouldn't be required anymore.

Reply:
Yes this makes sense @ivanprado , let me try this approach - I discarded this initially but now I see that performance-wise it should be very similar and looks simpler.

Reply:
Implemented in 6f2e2d2, looks much simpler indeed, thanks for suggestion @ivanprado